src/models/Member.model.ts

```diff
@@ -0,0 +1,20 @@
+import { prop } from '@typegoose/typegoose';
+
+export enum Role {
+  ORG_ADMIN = 'org_admin',
+  REGULAR = 'regular',
+}
+
+export class MemberModelSchema {
```

Comment:
Didn't we decide not to set any unique indexes?

Reply:
No, we considered enforcing uniqueness of email addresses and decided not to do it. We still need to enforce that any user name is unique in every organisation -- there can't be two `alice`s in `example.com`.

Heads-up in case you haven't done this before in MongoDB: Because the `userName` can be `null`, [you need a partial index](https://stackoverflow.com/questions/71276362/compound-index-where-one-field-can-be-null-mongodb), which [you can specify with Typegoose' `@index()`](https://typegoose.github.io/typegoose/docs/api/decorators/indexes/#example). I haven't tested the following but something like it should work:

```ts
@index({ orgName: 1 }, { partialFilterExpression: { userName: { $exists: true } } }) 
export class MemberModelSchema {
   // ...
}
```